src/PacketSerial.h

```diff
@@ -146,8 +162,14 @@ class PacketSerial_
 
     Stream* _serial;
 
-    PacketHandlerFunction _onPacketFunction;
-
+    struct {
+        bool hasBaton;
+        void* baton;
+        union {
+            PacketHandlerFunction function;
+            PacketHandlerFunctionWithBaton batonFunction;
+        };
```

Comment:
Note that storing this as a union and a flag results in less memory usage than storing both function pointers like you do in `master`